compiler/modules/types/list.bal

```diff
@@ -217,9 +217,16 @@ function listIntersectWith(FixedLengthArray members, SemType rest, ListAtomicTyp
     }
     int nonRepeatedLen = int:max(members.initial.length(), lt.members.initial.length());
     foreach int i in 0 ..< nonRepeatedLen {
-        fixedArraySet(members, i, 
-            intersect(listMemberAt(members, rest, i), listMemberAt(lt.members, lt.rest, i)));
-    } 
+        fixedArraySet(members, i, intersect(
+                                        listMemberAt(members, rest, i),
+                                        listMemberAt(lt.members, lt.rest, i)));
+    }
+    // Intersect the last member of initial array
+    if nonRepeatedLen < members.fixedLength {
+        members.initial[nonRepeatedLen] =  intersect(
+                                                listMemberAt(members, rest, nonRepeatedLen), 
```

Comment:
I typically don't format calls like this: I put the start of the first arg on the same line as the `(`